content/productivity/range-attack/plugin.js

```diff
@@ -0,0 +1,851 @@
+
+    setCursorMode(CursorMode.NoSelect);
+  }, [
+    setSourcePlanetLevels,
+    setSourcePlanetType,
+    setSourcePlanetEnergyPercent,
+    setSourcePlanetLevels,
+    setTargetPlanetType,
+    setOnlyArtifacts,
+    setSelectedSourcePlanet,
+    setSelectedTargetPlanet,
+  ]);
+
+  console.log("size", planetMoves.size);
+  const footer = html`
+    <div
+      style=${{
+        display: "flex",
+        flexDirection: "row",
+        alignItems: "center",
+        position: "absolute",
+        left: 0,
+        right: 0,
+        bottom: 0,
+        border: "0 solid white",
+        borderTopWidth: "1.5px",
+        padding: "16px 24px",
+      }}
+    >
+      <${CreateButton} ctaText="Attack!" onClick=${onAttack} />
+      <${CreateButton}
+        styles=${{
+          marginLeft: "4px",
+          marginRight: "4px",
+          borderWidth: 0,
+          textDecoration: "underline",
+        }}
+        ctaText="Reset Filters"
+        onClick=${resetFilters}
+      />
+      <div>${`capture ${planetMoves.size} planets`}</div>
+    </div>
+  `;
+
+  const selectedRectCoords = useCallback((planetRange) => {
+    if (planetRange?.beginCoords && planetRange?.endCoords) {
+      return `Begin:(${planetRange.beginCoords.x}, ${planetRange.beginCoords.y}), 
+          End:(${planetRange.endCoords.x}, ${planetRange.endCoords.y})`;
+    }
+    return "";
+  }, []);
+
+  console.log(selectedRectCoords(sourcePlanetRange));
+  return html`<div
+      style=${{
+        margin: "16px 4px 72px",
+      }}
+    >
+      <${createPlanetFilter}
+        planetLevels=${sourcePlanetLevels}
+        setPlanetLevels=${setSourcePlanetLevels}
+        planetType=${sourcePlanetType}
+        setPlanetType=${setSourcePlanetType}
+        planetEnergyPercentage=${sourcePlanetEnergyPercent}
+        setPlanetEnergyPercentage=${setSourcePlanetEnergyPercent}
+        cursorMode=${CursorMode.SelectSource}
+        selectedRectCoordsStr=${selectedRectCoords(sourcePlanetRange)}
+        selectedPlanet=${selectedSourcePlanet}
+        setSelectedPlanet=${setSelectedSourcePlanet}
+        title="Source Planet"
+      />
+      <div
+        style=${{
+          marginTop: "40px",
+        }}
+      ></div>
+      <${createPlanetFilter}
+        planetLevels=${targetPlanetLevels}
+        setPlanetLevels=${setTargetPlanetLevels}
+        planetType=${targetPlanetType}
+        setPlanetType=${setTargetPlanetType}
+        onlyArtifacts=${onlyArtifacts}
+        setOnlyArtifacts=${setOnlyArtifacts}
+        cursorMode=${CursorMode.SelectTarget}
+        selectedRectCoordsStr=${selectedRectCoords(targetPlanetRange)}
+        selectedPlanet=${selectedTargetPlanet}
+        setSelectedPlanet=${setSelectedTargetPlanet}
+        title="Target Planet"
+      />
+    </div>
+    ${footer} `;
+}
+
+export default class Plugin {
+  constructor() {
+    this.root = null;
+    this.container = null;
+
+    df.contractsAPI.contractCaller.queue.invocationIntervalMs = 50;
```

Comment:
Im not sure how I feel about mucking around with contracts api stuff. Ill have to think on that.
maybe its ok, but it would have to be documented and restored somehow upon destroy

Reply:
good suggestion!

Reply:
These can't be inside this plugin. The settings configured by the DF team are recommended by the xDai team directly. Additionally, the invocationIntervalMs was adjusted to 200ms earlier this week and no one has really been complaining since that change. Please remove these before acceptance 🙇